code/modules/projectiles/guns/energy/kinetic_accelerator.dm

```diff
@@ -144,10 +143,10 @@
 	overheat = FALSE
 
 /obj/item/gun/energy/kinetic_accelerator/update_icon()
-	..()
```

Comment:
Same here. These changes are not very reasonable long term.

Reply:
That's fine, I just felt our gun code had gone in this direction where the behavior is so diverse across weapons that there will never be a "universal" icon change and if it is it should be done in something like gun attachments where its handled only when it needs to be handled instead of making every gun check/rebuild its attachments every time it shoots/recharges/etc.